### ISSUE TITLE
ray init does not automaticly detect GPUs

### ISSUE THREAD

**Issue by alessiovignoli** [mathysgrapotte/stimulus#157]
ray is first initialized explicitly to set maximum resources limits, becuase it does not respect nextflow and cluster limits otherwise.

But doing so if num_gpus is left to None, default for ray.init see here -> [docs](https://docs.ray.io/en/latest/ray-core/api/doc/ray.init.html) the GPUs are not automaticly detected and used. In theory it should detect them all and use them all, but is not doing so.

**Comment by luisas** [comment]
@mathysgrapotte @alessiovignoli  Because now we do not use Ray anmore, should we close this one?

**Comment by mathysgrapotte** [comment]
yes ! I will add that this is now handled by nextflow requesting the proper type of ressources, if the instance where the process is running has gpus, those will be used, otherwise we throw a warning and run on CPU.
